src/convolution/convolution_2d.jl

```diff
@@ -0,0 +1,1153 @@
+
+This code basically goes over all the pixels in the image and applies the mathematical operation of convolution, that is the values of the pixel and its surrounding pixels are multipled by the value of the filter matrix. 
+
+However, it's alright if you don't understand exactly how the code works."""
+
+# ╔═╡ 44d594ff-a25b-4455-be56-863345c67b68
+function convolve(img, filter)
+    img_row, img_col = size(img)
+    filter_row, filter_col = size(filter)
+
+    filtered = zeros(img_row-filter_row+1, img_col-filter_col+1)
+    filtered_row, filtered_col = size(filtered)
+
+    for i in 1:filtered_row
+        for j in 1:filtered_col
+            filtered[i,j] = sum(img[i:i+filter_row-1,j:j+filter_col-1].*filter)
+        end
+    end
+  
+    return filtered
+end
+
+# ╔═╡ d752805d-ba2c-4133-a556-46a1da734557
+md"""Now let's start with a simple filter. This is called a "Box Blur" and it has the simple task of blurring the image.
+
+Can you see it ?"""
+
+# ╔═╡ d729f8b4-05e2-4863-b8c4-39b37646c36b
+test_filter = [0.111 0.111 0.111; 0.111 0.111 0.111; 0.111 0.111 0.111]
+
+# ╔═╡ 8ae0248f-6eba-4941-85f4-b21be3c7e725
+let
+	height, width = size(test_img)
+	filtered_img = Gray.(convolve(test_img, test_filter))
+	padded_img = test_img[2:height-1, 2:width-1]
+	[padded_img filtered_img]
+end
+
+# ╔═╡ 19b49665-0382-4eb1-9c70-8295e0aa819b
+md"""You can also try other filters! 
+
+Choose a filter from the selection below and see what the filter matrix looks like and how the image changes. 
+
+Can you identify the difference between the right and left shift matrices ?"""
+
+# ╔═╡ adc154cf-7059-4f1d-9bac-56b9a93cc47f
+@bind test_filter_selected Select([
+	[0 0 0; 0 1 0; 0 0 0] => "Original Image",
+	[1 2 1; 2 4 2; 1 2 1] * 0.0625 => "Gaussian Blur", 
+	[0.111 0.111 0.111; 0.111 0.111 0.111; 0.111 0.111 0.111] => "Box Blur",
+	[-0.111 -0.111 -0.111; -0.111 0.89 -0.111; -0.111 -0.111 -0.111] => "Details",
+	[-1 -1 -1; -1 8 -1; -1 -1 -1] => "Ridge Detection",
+	[0 -1 0; -1 4 -1; 0 -1 0] => "Edge Detection",
+	[1 2 1; 0 0 0; -1 -2 -1] => "Vertical Edges",
+	[1 0 -1; 2 0 -2; 1 0 -1] => "Horizontal Edges",
+	[0 0 0; 1 0 0; 0 0 0] => "Left Shift",
+	[0 0 0; 0 0 1; 0 0 0] => "Right Shift"])
+
+# ╔═╡ 6c31cbb0-01ed-40cf-b400-3ed6b7c79ecc
+test_filter_selected
+
+# ╔═╡ 4b560bf2-00d9-4440-9589-834cd9177f66
+begin
+	height, width = size(test_img)
+	filtered_img = Gray.(convolve(test_img, test_filter_selected))
+	padded_img = test_img[2:height-1, 2:width-1]
+	[padded_img filtered_img]
+end
+
+# ╔═╡ a174850b-91cc-4463-ab28-61ca0a7221c6
+md"""Now it's your turn! Try changing the values of the filter matrix and see what happens!"""
+
+# ╔═╡ f844bca2-edb7-45ec-b860-5f1e0b8b6bc0
+begin
+	#a = fill(0.,3,3)
+	a = [0.0 0.0 0.0; 0.0 1.0 0.0; 0.0 0.0 0.0]
+	#a[2,2] = 1.0
+	md"""**Exercice:** Can you find the right matrix to sharpen the image?"""
+end
+
+# ╔═╡ 2cf6f933-508a-4dce-b670-578afeb64519
+@bind filter Scrubbable(a,format="+2.1f")
```

Comment:
@fonsp here we run into the bug that the first time we get an error wth the Scrubabble matrix - anything we can do with it? Else it would mean everyone runing this notebook will immediately see an error :S